cpp/kiss_icp/pipeline/KissICP.cpp

```diff
@@ -59,24 +49,32 @@ KissICP::Vector3dVectorTuple KissICP::RegisterFrame(const std::vector<Eigen::Vec
     // Voxelize
     const auto &[source, frame_downsample] = Voxelize(cropped_frame);
 
-    // Get motion prediction and adaptive_threshold
-    const double sigma = GetAdaptiveThreshold();
+    // Get adaptive_threshold
+    const double sigma = adaptive_threshold_.ComputeThreshold();
 
     // Compute initial_guess for ICP
-    const auto prediction = GetPredictionModel();
-    const auto last_pose = !poses_.empty() ? poses_.back() : Sophus::SE3d();
-    const auto initial_guess = last_pose * prediction;
-
-    // Run icp
-    const Sophus::SE3d new_pose = registration_.AlignPointsToMap(source,         //
-                                                                 local_map_,     //
-                                                                 initial_guess,  //
-                                                                 3.0 * sigma,    //
-                                                                 sigma / 3.0);
+    const auto initial_guess = current_pose_ * current_delta_;
+
+    // Save current pose before the ICP loop to compute the current_delta_ later
+    const auto last_pose = current_pose_;
```

Comment:
Is this needed? `current_pose_` should not be changed until line 75 and before we already update `current_delta_`

Reply:
I Agree you benedetto on this, I am now renaming everything from `current_<things>` to `last_<things>`